SparkPost/matchbox
# UX-610 Convert ActionList to typescript

Note: `ts-loader` is extremely slow

### What Changed
- Converts `ActionList` and all its components to typescript
- Fixes `IconBase` types

### How To Test or Verify
- Verify the build works and type definitions for `ActionList` are generated
- Verify libby and the libby build work

### PR Checklist
